In [1]:
#firstly I will import the data by connecting to my google drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
# import the pacakages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import sklearn.linear_model as lm
import statsmodels.formula.api as smf
import statsmodels.api as sm
!pip install researchpy
import researchpy as rp


In [3]:
#Next I will load the dataset

HouseDF = pd.read_csv(datafile)
HouseDF.head()

Id  lotArea  quality  condition  BsmtSF  floorOneSF  floorTwoSF  \
0   1     8450        7          5     856         856         854   
1   2     9600        6          8    1262        1262           0   
2   3    11250        7          5     920         920         866   
3   4     9550        7          5     756         961         756   
4   5    14260        8          5    1145        1145        1053   

   bsmtFullBath  noRooms  firePlaces  garageCars  woodDeckSF  salePrice  
0             1        8           0           2           0     208500  
1             0        6           1           2         298     181500  
2             1        6           1           2           0     223500  
3             1        7           1           3           0     140000  
4             1        9           1           3         192     250000

In [4]:
#list(data) # to show column names get descriptive statistics etc
#list(data) # to show column names get descriptive statistics etc
HouseDF.describe()
list(HouseDF)
HouseDF.columns
HouseDF.count()
HouseDF.shape

(1460, 13)

These shows that: There are 1460 datapoints and 13 columns in our dataset.


##Objective

The objective of this analysis is to know the features of a house that will determine its price and also predict the price of a house based on the selected features using a multiple linear regression model

In [5]:
#next I will be dividing all of the columns in the dataset excluding the column_id to the target column and input column for further analysis
### identify the target  and input (dependent and independent variable )
target = 'salePrice'
inputDF = ['lotArea', 'quality', 'condition', 'BsmtSF', 'floorOneSF', 'floorTwoSF', 'bsmtFullBath', 'noRooms', 'firePlaces', 'garageCars', 'woodDeckSF']



In [6]:
# Fit the multiple regression model
X = HouseDF[inputDF]
y = HouseDF[target]
X = sm.add_constant(X) # add an intercept term
model = sm.OLS(y, X).fit() #this is caling out Ordinary Least Squares
model
# Print the results
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:              salePrice   R-squared:                       0.781
Model:                            OLS   Adj. R-squared:                  0.780
Method:                 Least Squares   F-statistic:                     470.3
Date:                Tue, 11 Jul 2023   Prob (F-statistic):               0.00
Time:                        21:41:41   Log-Likelihood:                -17434.
No. Observations:                1460   AIC:                         3.489e+04
Df Residuals:                    1448   BIC:                         3.496e+04
Df Model:                          11                                         
Covariance Type:            nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const        -1.209e+05   8030.981    -15.057   

The linear regression model has a coefficient of -120,930.5328, which represents the estimated mean value of the sales price when all other independent variables are set to zero.
Additionally, each independent variable has a positive coefficient indicating that an increase in that variable is associated with an increase in the estimated sales price, with all other variables held constant.
Specifically, the coefficients are as follows: 0.4019 for the square footage of the lot, 23,831.6846 for the quality of the house, 3,033.0293 for the condition of the house, 19.8187 for the square footage of the basement, 47.2156 for the first floor square footage, 41.0270 for the second floor square footage, 13,428.3661 for a basement with a full bath, 430.4322 for the number of rooms above ground, 4,869.3590 for the number of fireplaces, 17,243.3747 for the number of cars that can fit in the garage, and 34.8805 for the square footage of the wood deck.

The model used in this regression analysis is Ordinary Least Squares (OLS), and the model has a total of 1460 observations and 11 independent variables. With the independent variables, I was able to build a model a multiple linear regression model that predicts the price of a house based on the above variables. The P-value of the overall model is 0.000, which is less than 0.005 indicating that the model is statistically significant.
In the image above, the R-squared Value is high at 0.781, indicating that 78.1% of the variability in the Sales price is explained by the independent variables included in the model.


Based on the values above, the mathematical equation for the model is:

salePrice= -120920.5328 +0.4019lotArea+23831.6846quality + 3033.0293condition+19.8187BsmtSF+47.2156floorOneSF+41.0270floorTwoSF + 13428.3661bsmtFullBath+430.4322noRooms + 4869.3590firePlaces +17243.3747garageCars + 34.8805woodDeckSF  